metafix/src/test/java/org/metafacture/metafix/MetafixMethodTest.java

```diff
@@ -305,10 +305,32 @@ public void shouldNotTrimRepeatedField() {
         );
     }
 
+    @Test
+    public void shouldNotTrimIndexedArray() {
+        MetafixTestHelpers.assertExecutionException(IllegalStateException.class, "Expected String, got Hash", () ->
+            MetafixTestHelpers.assertFix(streamReceiver, Arrays.asList(
+                    "trim('data.title')"
+                ),
+                i -> {
+                    i.startRecord("1");
+                    i.startEntity("data");
+                    i.startEntity("title");
+                    i.literal("1", "  marc  ");
+                    i.literal("2", "  json  ");
+                    i.endEntity();
+                    i.endEntity();
+                    i.endRecord();
+                },
+                o -> {
+                }
+            )
+        );
+    }
+
     @Test
     // See https://github.com/metafacture/metafacture-fix/pull/133
     public void shouldNotTrimStringInImplicitArrayOfHashes() {
-        MetafixTestHelpers.assertExecutionException(IllegalStateException.class, "Expected String, got Array", () ->
+        MetafixTestHelpers.assertExecutionException(NumberFormatException.class, "For input string: \"title\"", () ->
```

Comment:
Hm, this doesn't look right. I think it should still be an `IllegalStateException`. Maybe "Expected Hash, got Array" then.

Reply:
Yes, I agree, addressed in 746fcf4.

Reply:
Thanks. I guess I would've placed it in `FixPath.InsertMode.REPLACE.apply(Array, String, Value)`. But doesn't matter too much at this point.

Reply:
Right, that's better. Pushed 5a3fa9b.